From e14ff320eda3ad36b5dd66d718326a48f715c0e6 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Mon, 7 Aug 2023 17:49:45 +0200
Subject: [PATCH] pinctrl: stm32: add data flag need_syscon

The pinctrl driver writes directly in the device specified by the
property 'st,syscfg' to program the interrupt mux.
The mux is inside the EXTI for MPUs and inside the st,stm32-syscfg
for the MCUs.

To prepare for EXTI to handle the mux directly, tag the pinctrl
for MCUs with the flag need_syscon; it indicated that the device
needs the property 'st,syscfg' to access st,stm32-syscfg. The MPU
devices don't need the property, but tolerate it for backward
compatibility with old DTs.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I5120c88e7504444bb774e10b7d8c3c59cda092aa
---
 drivers/pinctrl/stm32/pinctrl-stm32.h     | 1 +
 drivers/pinctrl/stm32/pinctrl-stm32f429.c | 1 +
 drivers/pinctrl/stm32/pinctrl-stm32f469.c | 1 +
 drivers/pinctrl/stm32/pinctrl-stm32f746.c | 1 +
 drivers/pinctrl/stm32/pinctrl-stm32f769.c | 1 +
 drivers/pinctrl/stm32/pinctrl-stm32h743.c | 1 +
 6 files changed, 6 insertions(+)

diff --git a/drivers/pinctrl/stm32/pinctrl-stm32.h b/drivers/pinctrl/stm32/pinctrl-stm32.h
index c368a24fae9f..26aba7c827d5 100644
--- a/drivers/pinctrl/stm32/pinctrl-stm32.h
+++ b/drivers/pinctrl/stm32/pinctrl-stm32.h
@@ -66,6 +66,7 @@ struct stm32_pinctrl_match_data {
 	bool secure_control;
 	bool io_sync_control;
 	bool rif_control;
+	bool need_syscon;
 };
 
 struct stm32_gpio_bank;
diff --git a/drivers/pinctrl/stm32/pinctrl-stm32f429.c b/drivers/pinctrl/stm32/pinctrl-stm32f429.c
index 663a97113c51..4f3f78c97bd7 100644
--- a/drivers/pinctrl/stm32/pinctrl-stm32f429.c
+++ b/drivers/pinctrl/stm32/pinctrl-stm32f429.c
@@ -1567,6 +1567,7 @@ static const struct stm32_desc_pin stm32f429_pins[] = {
 static struct stm32_pinctrl_match_data stm32f429_match_data = {
 	.pins = stm32f429_pins,
 	.npins = ARRAY_SIZE(stm32f429_pins),
+	.need_syscon = true,
 };
 
 static const struct of_device_id stm32f429_pctrl_match[] = {
diff --git a/drivers/pinctrl/stm32/pinctrl-stm32f469.c b/drivers/pinctrl/stm32/pinctrl-stm32f469.c
index ed5397f6d7f2..2def428eb193 100644
--- a/drivers/pinctrl/stm32/pinctrl-stm32f469.c
+++ b/drivers/pinctrl/stm32/pinctrl-stm32f469.c
@@ -1553,6 +1553,7 @@ static const struct stm32_desc_pin stm32f469_pins[] = {
 static struct stm32_pinctrl_match_data stm32f469_match_data = {
 	.pins = stm32f469_pins,
 	.npins = ARRAY_SIZE(stm32f469_pins),
+	.need_syscon = true,
 };
 
 static const struct of_device_id stm32f469_pctrl_match[] = {
diff --git a/drivers/pinctrl/stm32/pinctrl-stm32f746.c b/drivers/pinctrl/stm32/pinctrl-stm32f746.c
index d832aab7ebe1..c2319f927be9 100644
--- a/drivers/pinctrl/stm32/pinctrl-stm32f746.c
+++ b/drivers/pinctrl/stm32/pinctrl-stm32f746.c
@@ -1662,6 +1662,7 @@ static const struct stm32_desc_pin stm32f746_pins[] = {
 static struct stm32_pinctrl_match_data stm32f746_match_data = {
 	.pins = stm32f746_pins,
 	.npins = ARRAY_SIZE(stm32f746_pins),
+	.need_syscon = true,
 };
 
 static const struct of_device_id stm32f746_pctrl_match[] = {
diff --git a/drivers/pinctrl/stm32/pinctrl-stm32f769.c b/drivers/pinctrl/stm32/pinctrl-stm32f769.c
index f81c51c5aedf..756494fccb83 100644
--- a/drivers/pinctrl/stm32/pinctrl-stm32f769.c
+++ b/drivers/pinctrl/stm32/pinctrl-stm32f769.c
@@ -1802,6 +1802,7 @@ static const struct stm32_desc_pin stm32f769_pins[] = {
 static struct stm32_pinctrl_match_data stm32f769_match_data = {
 	.pins = stm32f769_pins,
 	.npins = ARRAY_SIZE(stm32f769_pins),
+	.need_syscon = true,
 };
 
 static const struct of_device_id stm32f769_pctrl_match[] = {
diff --git a/drivers/pinctrl/stm32/pinctrl-stm32h743.c b/drivers/pinctrl/stm32/pinctrl-stm32h743.c
index ffe7b5271506..f18907d1c858 100644
--- a/drivers/pinctrl/stm32/pinctrl-stm32h743.c
+++ b/drivers/pinctrl/stm32/pinctrl-stm32h743.c
@@ -1959,6 +1959,7 @@ static const struct stm32_desc_pin stm32h743_pins[] = {
 static struct stm32_pinctrl_match_data stm32h743_match_data = {
 	.pins = stm32h743_pins,
 	.npins = ARRAY_SIZE(stm32h743_pins),
+	.need_syscon = true,
 };
 
 static const struct of_device_id stm32h743_pctrl_match[] = {
-- 
2.39.5

